File: services/origin-service/json_response.php
<?php
// json_response.php
// Include this helper and use json_ok/json_error instead of echo


// --- Unified JSON Response Functions ---
function json_ok($data = null, $message = "Success") {
    header('Content-Type: application/json');
    echo json_encode([
        "status"  => "ok",
        "message" => $message,
        "data"    => json_decode($data)
    ], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
    exit;
}

function json_error($message, $code = 500) {
    header('Content-Type: application/json');
    http_response_code($code);
    echo json_encode([
        "status"  => "error",
        "message" => $message
    ], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
    exit;
}

function json_pending($message = "Pending") {
    header('Content-Type: application/json');
    http_response_code(202);
    echo json_encode([
        "status"  => "pending",
        "message" => $message
    ], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
    exit;
}

function json_unauthorized($message = "Invalid Login") {
    header('Content-Type: application/json');
    http_response_code(401);
    echo json_encode([
        "status"  => "unauthorized",
        "message" => $message
    ], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
    exit;
}

?>

File: services/kafka-service/src/main/java/com/hmdev/messaging/service/kafka/data/JsonResponse.java
package com.hmdev.messaging.service.kafka.data;

public class JsonResponse {
    public String message;
    public Object data;
    public String status;

    private JsonResponse() {
    }

    public static JsonResponse ok() {
        return ok(null);
    }

    public static JsonResponse ok(Object data) {
        JsonResponse response = new JsonResponse();
        response.status = "ok";
        response.data = data;
        return response;
    }

    public static JsonResponse error(Object msg) {
        JsonResponse response = new JsonResponse();
        response.status = "error";
        response.message = msg.toString();
        return response;
    }

    public static JsonResponse unauthorized(String msg) {
        JsonResponse response = new JsonResponse();
        response.status = "unauthorized";
        response.message = msg;
        return response;
    }
}
